src/data_structures/src/tests/bit_array_test.cairo

```diff
@@ -294,6 +316,8 @@ fn test_serde_serialize() {
     let length = out.pop_front().unwrap();
     let length: usize = length.try_into().unwrap();
     assert!(length == ba.len(), "len not equal");
+    // We gotta skip one now
+    out.pop_front().unwrap();
```

Comment:
why ?

Reply:
maybe I can do so it doesn't change, lmk

Reply:
what's the change ? sounds confusing to have a useless element at the begining

Reply:
just used the generic array serde instead of writing my own. Will look back into it

Reply:
ah yes no it makes sense it's the length of the array no ?

Reply:
Yes

Reply:
cool let's leave it as is

Reply:
for some reason len is different than the array size...
I didn't try to understand, just updating the code

Reply:
ok good enough